src/main/java/org/truffleruby/language/arguments/keywords/KeywordDescriptorManager.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+package org.truffleruby.language.arguments.keywords;
+
+import org.truffleruby.RubyLanguage;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public class KeywordDescriptorManager {
+
+    public static final KeywordDescriptorManager INSTANCE = new KeywordDescriptorManager();
+
+    private final ConcurrentHashMap<NonEmptyKeywordDescriptor, NonEmptyKeywordDescriptor> CANONICAL_DESCRIPTORS = new ConcurrentHashMap<>();
```

Comment:
This caches RubySymbol in `static final`, that's incorrect, we need 1 KeywordDescriptorManager per RubyLanguage instance. (or do not store anything language instance-specific in NonEmptyKeywordDescriptor)